index/rolodex.go

```diff
@@ -425,6 +426,26 @@ func (r *Rolodex) BuildIndexes() {
 	r.manualBuilt = true
 }
 
+// GetAllReferences  returns all references found in the root and all other indices
+func (r *Rolodex) GetAllReferences() map[string]*Reference {
+	allRefs := make(map[string]*Reference)
+	for _, idx := range append(r.GetIndexes(), r.GetRootIndex()) {
```

Comment:
@daveshanley I made these methods because using GetIndexes() alone was not enough. It seems that GetIndexes() does not include the root index. Is that on purpose? Rather than returning the root index in GetIndexes(), I added these methods, just to be safe.